python/lsst/meas/base/wrappers.py

```diff
@@ -442,3 +444,201 @@ def _init(self, ctrl, name, mapper, logName=None):
+
+    def measure(self, measRecord, exposure, center):
+        """Measure source
+
+        It is the responsibility of this method to perform the desired
+        measurement and record the result in the `measRecord`.
+
+        Parameters
+        ----------
+        measRecord : `lsst.afw.table.SourceRecord`
+            Catalog record for the source being measured.
+        exposure : `lsst.afw.image.Exposure`
+            Exposure on which the source is being measured.
+        center : `lsst.afw.geom.Point2D`
+            Pixel coordinates of the object.
+
+        Raises
+        ------
+        `MeasurementError`
+            If the measurement fails for a known/justifiable reason.
+        """
+        raise NotImplementedError()
+
+    def measureN(self, measCat, exposure, refCat, refWcs):
+        """Measure multiple sources
+
+        It is the responsibility of this method to perform the desired
+        measurement and record the result in the `measCat`.
+
+        Parameters
+        ----------
+        measCat : `lsst.afw.table.SourceCatalog`
+            Catalog for the sources being measured.
+        exposure : `lsst.afw.image.Exposure`
+            Exposure on which the source is being measured.
+        refCat : `lsst.afw.table.SourceCatalog`
+            Reference catalog.
+        refWcs : `lsst.afw.image.Wcs`
+            Astrometric solution for the reference image.
+
+        Raises
+        ------
+        `MeasurementError`
+            If the measurement fails for a known/justifiable reason.
+        """
+        raise NotImplementedError()
+
+    def fail(self, measRecord, error=None):
+        """Record failure
+
+        This default implementation simply records the failure
+        in the source record.
+
+        Parameters
+        ----------
+        measRecord : `lsst.afw.table.SourceRecord`
+            Catalog record for the source being measured.
+        error : `Exception`
+            Error causing failure, or `None`.
+        """
+        measRecord.set(self._failKey, True)
+
+
+def singleFramePluginFromGeneric(name, GenericPluginClass):
+    """Produce a SingleFramePlugin sub-class from a GenericPlugin class
+
+    The class is also registered.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of plugin to register.
+    GenericPluginClass : `type`, subclass of `GenericPlugin`
+        `GenericPlugin` subclass to wrap.
+    """
+    class SingleFrameFromGenericConfig(GenericPluginClass.ConfigClass, SingleFramePluginConfig):
+        pass
+
+    @register(name)
+    class SingleFrameFromGenericPlugin(SingleFramePlugin):
+        ConfigClass = SingleFrameFromGenericConfig
+
+        def __init__(self, config, name, schema, metadata, logName=None):
+            SingleFramePlugin.__init__(self, config, name, schema, metadata, logName=logName)
+            self._generic = GenericPluginClass(config, name, schema, metadata)
```

Comment:
I am nearly positive that if you are creating these classes inside a function the GenericPluginClass (or cls; see below) variable will be in a closure, and there is no need to explicitly store another pointer to it inside the self object.

Reply:
I'm storing an instance, not the class itself. I don't think we want to instantiate it every time we want to use it.
Or have I misunderstood your point?